### DISCUSSION TITLE
Issue with external redirect on native app (iOS)

### DISCUSSION THREAD

**Discussion by treywhite21** [nextauthjs/next-auth#7446] [Help]
### Environment

System:
    OS: macOS 12.2.1
    CPU: (8) arm64 Apple M1
    Memory: 145.16 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh

  Binaries:
    Node: 19.5.0 - /opt/homebrew/bin/node
    npm: 9.3.1 - /opt/homebrew/bin/npm

  Browsers:
    Chrome: 112.0.5615.137
    Firefox: 107.0
    Safari: 15.3

### Reproduction URL

done-deal.vercel.app

### Describe the issue

I created a NextJS app (using app directory, if that matters). I set up next-pwa and then followed instructions at pwabuilder.com to deploy to iOS app (only test flighting at the moment).

On my iPhone, if I "install" the app directly as a PWA through Safari "save to home screen," Google authentication is working as expected. However, when using the actual iOS application, the external redirect to select Google accounts caused an error where the splash screen is shown with a blinking wifi icon with a slash through it. It's never able to "leave" the app to allow Google authentication to proceed.

Is it possible to use next-auth with a dedicated native application?

### How to reproduce

Explained in issue description.

### Expected behavior

Expected behavior would be using the iOS app and being able to sign in using Google authentication via the same mechanisms as I can do when using the PWA version.

**Comment by MarianDabrowski** [comment]
Thanks @treywhite21 for creating this issue and starting the discussion. 
BTW: any updates? 

I have the same issue. I wonder what is the root cause of this issue.  I confirm that the next-auth login work on android and web, but iOS follows the exact problem that is described above.

**Comment by LedzZm** [comment]
Hi, are you aware of WKAppBoundDomains ? Basically WebKit dissallows in browser redirects outside of the app's domain so that external site's can not track user interactions in the app.
Maybe I am not explaining it very well but in any case try to include an array of the login redirect links to your WKAppBoundDomains array in the Info.plist file of the compiled app (It should be in the src of the pwabuilder export)
like so 
```
    <key>WKAppBoundDomains</key>
    <array>
        <string>your.app.domain</string>
        <string>appleid.apple.com</string>
        <string>accounts.google.com </string>
        <string>www.facebook.com</string>
    </array>
```
Let me know if it worked

**Reply by LedzZm** [reply]
My problem is that I don't get redirected back to the app (Just see a white screen)
This happens only on ios devices
using browsers (Safari included) It works as expected, as well as with the app downloaded from GPlay store
only in ios devices it does not work properly

**Reply by MarianDabrowski** [reply]
I confirm that it this solution works and it is not a next-auth issue. You just need to configure it correctly!

**Reply by MarianDabrowski** [reply]
Yet I do not have the white screen problem

**Reply by LedzZm** [reply]
Yes the white screen issue is not next-auth related, I was using a new tab for the login, which is not handled in iOS.
Since the solution are the app bound domains we could mark this as answered I guess.
